pkg/apis/pulumi/v1alpha1/stack_types.go

```diff
@@ -108,6 +116,32 @@ type StackSpec struct {
 	RetryOnUpdateConflict bool `json:"retryOnUpdateConflict,omitempty"`
 }
 
+// GitAuthConfig specifies git authentication configuration options.
+// There are 3 different authentication options:
+//   * Personal access token
+//   * SSH private key (and it's optional password)
+//   * Basic auth username and password
+// Only 1 authentication mode is valid.
+type GitAuthConfig struct {
+	PersonalAccessToken *ResourceRef `json:"accessToken,omitempty"`
+	SSHAuth             *SSHAuth     `json:"sshAuth,omitempty"`
+	BasicAuth           *BasicAuth   `json:"BasicAuth,omitempty"`
```

Comment:
Is the casing on `json:"BasicAuth,omitempty"` correct?

Reply:
Fixed.